Explorer/src/components/sidebar.tsx

```diff
@@ -0,0 +1,118 @@
+"use client";
+
+import classNames from "classnames";
+import { Sidebar as FlowbiteSidebar } from "flowbite-react";
+import type { FC, PropsWithChildren } from "react";
+import { useSidebarContext } from "@/context/sidebar_context";
+import {
+  HiPresentationChartLine,
+  HiCog,
+  HiFolder,
+  HiServer,
+  HiTemplate,
+  HiChip,
+  HiTable,
+  HiHome,
+} from "react-icons/hi";
+
+const Sidebar: FC<PropsWithChildren<Record<string, unknown>>> = function ({
+  children,
+}) {
+  const { isOpenOnSmallScreens: isSidebarOpenOnSmallScreens } =
+    useSidebarContext();
+
+  return (
+    <div
+      className={classNames(
+        "fixed overflow-auto top-0 h-screen z-10 lg:sticky lg:!block",
+        {
+          hidden: !isSidebarOpenOnSmallScreens,
+        }
+      )}
+    >
+      <FlowbiteSidebar>{children}</FlowbiteSidebar>
+    </div>
+  );
+};
+
+/*import { useNavigationStore } from "@/context/sidebar_context";*/
+
+export default Object.assign(Sidebar, { ...FlowbiteSidebar });
+
+export function ActualSidebar(): JSX.Element {
+  const current_page = "test"; /*useNavigationStore((state) => state.page);*/
```

Comment:
I think we should do this this using [getPathname](https://nextjs.org/docs/app/api-reference/functions/use-pathname)
As this is a decision entirely contained within the sidebar it should not use shared state.